src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -158,155 +156,146 @@ private String getTenant(Map<String, String> headers) {
     return TenantTool.calculateTenantId(headers.get(OKAPI_TENANT_HEADER));
   }
 
+  private String buildUserLookupURL(String okapiURL, String username, String userId) {
+    String requestURL;
+    if(username != null) {
+      requestURL = String.format("%s/users?query=username==%s", okapiURL,
+          StringUtil.urlEncode(username));
+    } else {
+      requestURL = String.format("%s/users?query=id==%s", okapiURL,
+          StringUtil.urlEncode(userId));
+    }
+
+    return requestURL;
+  }
+
+  private JsonObject extractUserFromLookupResponse(HttpResponse<Buffer> res, String requestURL, String username) throws UserLookupException {
+    if (res.statusCode() != 200) {
+      String message = "Error looking up user at url '" + requestURL + "' Expected status code 200, got '" + res.statusCode()
+          + "' :" + res.bodyAsString();
+      throw new UserLookupException(message);
+    } else {
+      JsonObject lookupResult = res.bodyAsJsonObject();
+      if (!lookupResult.containsKey("totalRecords") || !lookupResult.containsKey("users")) {
+        throw new UserLookupException("Error, missing field(s) 'totalRecords' and/or 'users' in user response object");
+      } else {
+        int recordCount = lookupResult.getInteger("totalRecords");
+        if (recordCount > 1) {
+          throw new UserLookupException("Bad results from username");
+        } else if (recordCount == 0) {
+          throw new UserLookupException("No user found by username " + username);
+        }
+      }
+      return lookupResult.getJsonArray("users").getJsonObject(0);
+    }
```

Comment:
Using guard clauses avoids all else clauses and indentation resulting in better readability.
```suggestion
    if (res.statusCode() != 200) {
      String message = "Error looking up user at url '" + requestURL + "' Expected status code 200, got '" + res.statusCode()
          + "' :" + res.bodyAsString();
      throw new UserLookupException(message);
    }
    JsonObject lookupResult = res.bodyAsJsonObject();
    if (!lookupResult.containsKey("totalRecords") || !lookupResult.containsKey("users")) {
      throw new UserLookupException("Error, missing field(s) 'totalRecords' and/or 'users' in user response object");
    }
    int recordCount = lookupResult.getInteger("totalRecords");
    if (recordCount > 1) {
      throw new UserLookupException("Bad results from username");
    }
    if (recordCount == 0) {
      throw new UserLookupException("No user found by username " + username);
    }
    return lookupResult.getJsonArray("users").getJsonObject(0);
```